lib/jxl/encode.cc

```diff
@@ -401,6 +409,12 @@ JxlEncoderStatus JxlEncoderAddImageFrame(const JxlEncoderOptions* options,
     c_current = options->enc->metadata.m.color_encoding;
   }
 
+  // treat multiple frames as animation
+  // until animation API is established
+  options->enc->metadata.m.have_animation = true;
```

Comment:
Shouldn't this be set to true only if this is not the first frame? Or is there something else that ignores the have_animation if there's only one frame? I think the same would apply to duration of previous frames.

Reply:
@xiota not just saying that if there's only one frame it shouldn't have_animation = true. Lode's point that this should actually be set by the basic info is relevant here.

Reply:
@deymo I missed that basic info already has an animation property.  I'll change it to set the duration based on the setting from basic info.  What I don't understand now is how the API should behave when it receives multiple frames to encode and animation is set to false.  Currently, non-animation, multiframe codestreams are undecodable by `djxl`.  Should the API refuse to encode multiple frames when animation is not set (until `djxl` and the decode API are able to handle such codestreams)?

Reply:
Either we return an error the second time you add a frame if have_animation is false; or we take that as a way to set multiple frames in a still image (you can define multiple internal JXL "frames" that are all meant to be blended in the decoder, like layers in photoshop). We can return an error for now and work on the "layer" support (both encoder and decoder) later. There are many blending modes available to internal frames so this is not as simple as adding multiple frames in the encoder api anyway.

Reply:
Note that the issue is not that `djxl` doesn't understand those streams; we are most likely encoding something wrong in the encoder side in that case related to boxes. `djxl` is meant to blend all the frames of a still image (have_animation=false).